examples/minimal_cxx_qt_qml/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+// SPDX-FileCopyrightText: 2021 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+// SPDX-FileContributor: Gerhard de Clercq <gerhard.declercq@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+use cxx_qt::make_qobject;
+
+#[make_qobject]
+mod my_object {
+    #[derive(Default)]
+    pub struct Data {
+        number: i32,
+        string: String,
+    }
+
+    #[derive(Default)]
+    struct RustObj;
+
+    impl RustObj {
+        #[invokable]
+        fn increment_number(&self, cpp: Pin<&mut CppObj>) {
+            let value = cpp.number() + 1;
```

Comment:
Wonder if this should use the wrapper instead, as that's what will happen in the long term. As for that specific case you can get away with using the CppObj, but as soon as you try to do multiple write changes you'll get compile issues.

A wrapper route might look like this

```rust
let mut wrapper = CppObjWrapper::new(cpp);
let value = wrapper.number() + 1;
wrapper.set_number(value);
```

Eventually this would be transparent to the user though and instead of `Pin<&mut CppObj>` they would have a `CppObjectWrapper` etc.

Reply:
So maybe it's OK to leave as you have currently. As eventually the user won't know there is a wrapper.

And maybe this indicates this task needs to be done soonish :-)

Reply:
This will be done in #8

Reply:
Ah, I didn't know that could cause such problems, I'll update it to the Wrapper type.
If we want to demonstrate that code, it better use the appropriate calls.

Reply:
Fixed